src/event.rs

```diff
@@ -87,6 +88,8 @@ pub enum Event {
     /// Because of repeat, there may be a number `KeyDown` events before
     /// a corresponding `KeyUp` is sent.
     KeyUp(KeyEvent),
+    /// Called when a paste command is received.
```

Comment:
Can you briefly explain why this needs to be its own event and not treated as an instance of command? Part of the reason I ask is that it's going to grow variants ("paste without formatting") and so this feels a bit like a layering violation.

Reply:
This was an experiment. `Command` is a funny thing. My gut feeling is that 'command' is the solution for when we need runtime dynamism, and are willing to sacrifice ergonomics. In the case of paste, event flow works and is more explicit.

I'm not sure this is correct, of course.

In the case of 'paste' growing options, I think that's okay; we could have the argument be a `PasteEvent` that has both `ClipboardItem` and `PasteOptions` fields.

### About `Command`s:

`Command` is a bit funny. Per my current understanding, `Command` has two main uses; the first is to allow passing of arbitrary messages around the tree, and the second is to enable certain standard platform actions, usually related to menus.

Here's an example of how this is a bit strange:

When the user selects the "new window" item in a menu, the system sends a `NEW_WINDOW` command up the widget tree. This command doesn't actually create the new window, though, because the menu doesn't know the details of the window the user wants to create; instead the user is expected to receive this command and then submit _another_ command, `CREATE_WINDOW`, which actually has as a payload the window we want to create.

I don't see a good alternative to Commands for the first of these cases, the command we get from the menu. But I _could_ imagine `CREATE_WINDOW` being replaced by some method that we expose on `WinCtx`, or elsewhere.

So this is the general question: what sorts of things should be commands, which should be events, and which should be methods? Is there a good test we can apply to decide?

Reply:
Hmm. The criterion I was (probably subconsciously) applying is that event is "comes from system" which "create window" unambiguously falls under, while command is "an app concern," which is also related to how open-ended it is. But this might not be the right way to slice it.

I'm willing to accept this PR as "good enough" and try and figure out the details later. It's possible my concerns would be settled by having an extension point in the `Event` enum that could be filled out by the app.

Reply:
So _my_ thinking here is that an event is something that "comes from the framework"; and things like paste are a bit of a gray zone; there are certainly things very similar to paste, like drag and drop, that are unambiguously system events.

As a hiccup for "comes from system": we will want lifecycle events at some point, and while those are related to things happening in the system they're also fairly artificial; I would imagine having an event immediately after a widget is added to the tree, for instance, and that wouldn't be coming from the system at all?

Anyway: this is not to say that this approach is good, just further explication of my thinking.

Reply:
What you say makes total sense, and by "system" I'm not trying to make a sharp distinction with framework events. For example, we already have "HotChanged" in that category (synthesized by system, happens to be derived from mouse move event from system), and I'm not suggesting moving that out.

Total agreement that drag'n'drop belongs in event. I think we basically agree that "paste" is in the gray zone here, it *could* be modeled purely as an app command, the handling of which would be doing a (synchronous?) query into the platform to retrieve the paste buffer, but it's not necessary to do so.

So basically, yeah, this is fine for now and not urgent to reconsider.